linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1187,16 +1187,22 @@ private[emitter] object CoreJSLib {
         val zero = varRef("zero")
         val arrayEncodedName = varRef("arrayEncodedName")
         val displayName = varRef("displayName")
-        val isArrayOf = varRef("isArrayOf")
-        val obj = varRef("obj")
+        val self = varRef("self")
+        val that = varRef("that")
         val depth = varRef("depth")
+        val obj = varRef("obj")
         MethodDef(static = false, Ident("initPrim"),
-            paramList(zero, arrayEncodedName, displayName, isArrayOf), {
+            paramList(zero, arrayEncodedName, displayName), {
           Block(
               privateFieldSet("ancestors", ObjectConstr(Nil)),
               privateFieldSet("zero", zero),
               privateFieldSet("arrayEncodedName", arrayEncodedName),
-              privateFieldSet("isArrayOf", isArrayOf),
+              const(self, This()),
+              privateFieldSet("isAssignableFromFun", {
+                Function(arrow = false, paramList(that), {
+                  Return(that === self)
```

Comment:
I do not understand why we cannot rely on `This()` in this function. It seems it is always called with a receiver.